### PR TITLE
Remove unused dependencies + fix some lint config (2)

### PR DESCRIPTION
**Pull request by webpro**
Basically a duplicate of #507. After `git fetch` this gives no diff:

```
git diff chore/remove-unused-dependencies..chore/remove-unused-dependencies-2 -- . ':(exclude)common/config/rush/pnpm-lock.yaml'
```

So there are only changes in the lockfile (not even in the repo state file).

The error in the other branch that does not reveal in this one: https://kadena-io.slack.com/archives/C04AL4NRWGP/p1688465334175419

I find this very odd :)

Will _greenify CI_ by updating the changelog.

### CODE REVIEW

**Review by Randynamic — APPROVED** [review]

File: packages/libs/bootstrap-lib/package.json
Comment (alber70g) [inline packages/libs/bootstrap-lib/package.json:1]: Can we change the `.eslintrc` to use `@kadena-dev/eslint-config....lib` as `extends` option?

File: packages/apps/graph/package.json [hunk 1]
```diff
@@ -44,7 +43,6 @@
     "@types/node": "^16.0.0",
     "embedded-postgres": "~15.1.0-beta.6",
     "prettier": "~2.8.8",
-    "sort-package-json": "~2.4.1",
-    "ts-node": "~10.8.2"
+    "sort-package-json": "~2.4.1"
```
Comment (alber70g) [inline packages/apps/graph/package.json:46]: We can replace `sort-package-json` with https://github.com/kellyselden/eslint-plugin-json-files/blob/master/docs/rules/sort-package-json.md

File: packages/libs/kadena.js/.eslintrc.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // This is a workaround for https://github.com/eslint/eslint/issues/3458
-require('@kadena-dev/eslint-config/profile/lib');
+require('@rushstack/eslint-config/patch/modern-module-resolution');
```
Comment (alber70g) [inline packages/libs/kadena.js/.eslintrc.js:2]: Why was this changed?

**Review by alber70g — APPROVED** [review]
Question: can we also introduce linting for unused dependencies in this PR?

**Comment by webpro** [comment]
> Question: can we also introduce linting for unused dependencies in this PR?

That's quite a thing. This monorepo has a lot of difficult to find unused dependencies, so even the mighty Knip lib has troubles and reports false positives. And it also requires a `package.json` in the root of the repo which we don't have. So I'm working on this, but the story isn't trivial.

**Review by jermaine150 — APPROVED** [review]

**Comment by alber70g** [comment]
> > Question: can we also introduce linting for unused dependencies in this PR?
> 
> That's quite a thing. This monorepo has a lot of difficult to find unused dependencies, so even the mighty Knip lib has troubles and reports false positives. And it also requires a `package.json` in the root of the repo which we don't have. So I'm working on this, but the story isn't trivial.

Fair enough. What about having a whitelist of dependencies that shouldn't be touched, and a `--verify`? Something for the future I reckon

File: packages/libs/bootstrap-lib/package.json
Comment (webpro) [reply packages/libs/bootstrap-lib/package.json:1]: Turns out this opens up a can of worms, so I'll pick this up after merging this one.
